Added a Gitlab CI automation (#19)
AP-Sensing/PhotonPonyOS: README.md

 
After successfully building, the image will be located in `iso/linux/images/boot.iso`.
This image then can be flashed to a USB-stick with for example the [Fedora Media Writer](https://flathub.org/apps/org.fedoraproject.MediaWriter).

### Gitlab CI automation

PPOS can be build in a Gitlab-Runner using a Docker-Executor.

> The loop module has to be loaded on the Host of the GitLab-Runner

Reviewer: ```suggestion
The loop module has to be loaded on the host of the GitLab-Runner.
This can be done via:
...

The GitLab tomel file (path to the file) also needs no be changed and `privileged` mode needs to be enabled.
Why? Because XYZ
```

Please add documentation for how to load it.

Author: Spelling corrections and explanation done

```toml
  [runners.docker]
    image = "fedora:38"
    privileged = true

> ```

Reviewer: Is this all that is needed for building it inside a gitlab runner?

Author: The rest is private information like url, id, token etc.

> 
> 

Reviewer: ```suggestion

```

 
After successfully building, the image will be located in `iso/linux/images/boot.iso`.
This image then can be flashed to a USB-stick with for example the [Fedora Media Writer](https://flathub.org/apps/org.fedoraproject.MediaWriter).

### Gitlab CI automation

PPOS can be build in a Gitlab-Runner using a Docker-Executor.
The loop module has to be loaded on the host of the GitLab-Runner. This can be done via:
```bash
sudo modprobe loop
```

> The Gitlab configuration file can be found at /etc/gitlab-runner/config.toml

Reviewer: ```suggestion
The Gitlab configuration file can be found at `/etc/gitlab-runner/config.toml`.
```

```toml
  [runners.docker]
    image = "fedora:38"
    privileged = true
```

> Since building the image requires read/write permissions on mounted devices the docker needs to be privileged.

Reviewer: ```suggestion
Since building the image requires read/write permissions on mounted devices (e.g. `/dev/loop0`), docker images using it need to be marked as `privileged`.
```
